File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int max_sum = nums[0]; 
        int curr_sum = nums[0]; 
        for(int i = 1; i < nums.size(); i++){
            int includeSum = nums[i] + curr_sum; 
            max_sum = max(max_sum, max(includeSum, nums[i])); 
            curr_sum = max(includeSum, nums[i]); 
        }
        return max_sum; 
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int answer = nums[0]; 
        int curr = nums[0]; 
        for(int i = 1; i < nums.length; i++){
            int sum = curr + nums[i]; 
            curr = Math.max(nums[i],sum); 
            answer = Math.max(answer, curr); 
        }
        return answer; 
    }
}